File: Array/3. secondSmallest.ts
function findSecondSmallest(arr: number[]) {
  let smallest = Infinity;
  let secondSmallest = Infinity;

  if (arr.length < 2) {
    return -1;
  }

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] < smallest) {
      secondSmallest = smallest;
      smallest = arr[i];
    } else if (arr[i] < secondSmallest && arr[i] !== smallest) {
      secondSmallest = arr[i];
    }
  }
  return secondSmallest;
}

console.log(findSecondSmallest([1, 2, 4, 7, 7, 5]));
console.log(findSecondSmallest([1]));

File: Array/3. secondSmallest.py
def optimal_solution(arr):
    n = len(arr)
    smallest = arr[0]
    s_smallest = float("inf")
    for i in range(n):
        if arr[i] < smallest:
            s_smallest = smallest
            smallest = arr[i]
        elif arr[i] < s_smallest and arr[i] != smallest:
            s_smallest = arr[i]

    return s_smallest


print(optimal_solution([1, 2, 4, 7, 7, 5]))
